mongo/integration/unified/collection_operation_execution.go

```diff
@@ -488,6 +536,58 @@ func executeFind(ctx context.Context, operation *operation) (*operationResult, e
 	return newCursorResult(docs), nil
 }
 
+func executeFindOne(ctx context.Context, operation *operation) (*operationResult, error) {
+	coll, err := entities(ctx).collection(operation.Object)
+	if err != nil {
+		return nil, err
+	}
+
+	var filter bson.Raw
+	opts := options.FindOne()
+
+	elems, _ := operation.Arguments.Elements()
+	for _, elem := range elems {
+		key := elem.Key()
+		val := elem.Value()
+
+		switch key {
+		case "collation":
+			collation, err := createCollation(val.Document())
+			if err != nil {
+				return nil, fmt.Errorf("error creating collation: %v", err)
+			}
+			opts.SetCollation(collation)
+		case "filter":
+			filter = val.Document()
+		case "hint":
+			hint, err := createHint(val)
+			if err != nil {
+				return nil, fmt.Errorf("error creating hint: %v", err)
+			}
+			opts.SetHint(hint)
+		case "maxTimeMS":
+			opts.SetMaxTime(time.Duration(val.Int32()) * time.Millisecond)
+		case "projection":
+			opts.SetProjection(val.Document())
+		case "sort":
+			opts.SetSort(val.Document())
+		default:
+			return nil, fmt.Errorf("unrecognized findOne option %q", key)
+		}
+	}
+	if filter == nil {
+		return nil, newMissingArgumentError("filter")
+	}
+
+	res, err := coll.FindOne(ctx, filter, opts).DecodeBytes()
+	// Ignore ErrNoDocuments errors from DecodeBytes.
+	if err == mongo.ErrNoDocuments {
```

Comment:
`executeFindOne`, `executeFindOneAndDelete`, `executeFindOneAndReplace` and `executeFindOneAndUpdate` were all failing when there were no documents found, as `DecodeBytes` returns an `mongo.ErrNoDocuments` in this case. I've explicitly set the error to `nil` in this case, as we have previously suggested users do.

Reply:
Ah elaborated on the comment(s) a bit more. The documentation for [`DecodeBytes`](https://github.com/mongodb/mongo-go-driver/blob/master/mongo/single_result.go#L78) does call out the returning of `ErrNoDocuments` when no documents are present in the underlying cursor.